lib/scenic/theme.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Scenic.Theme do
+  @moduledoc """
+  A Theme is a collection of settings for things like font, color pallettes, and so forth.
+  """
+  defstruct font: nil, pallette: %{}, name: nil
+
+  @type pallette :: %{
+          atom() => Scenic.Color.t()
+        }
+
+  @typedoc """
+  A Theme is a collection of settings for things like font, color pallettes, and so forth.
+
+  The available themeable properties are:
+
+  * `name` is the atom identifying a theme.
+  * `font` is a String of the font that a theme should use, or nil if no preference.
+  * `pallette` is a map mapping atoms like `:primary` or `:secondary` to a `Scenic.Color`.
+  """
+  @type t :: %__MODULE__{
+          name: :atom,
+          font: String.t() | nil,
+          pallette: pallette()
+        }
+
+  @doc """
+  Creates a theme object with a given name.
+  """
+  @spec create(any) :: {:ok, t()} | {:error, String.t()}
+  def create(name) when is_atom(name), do: {:ok, %__MODULE__{name: name}}
```

Comment:
We also discussed having a version of this that would take a bulk update of settings (similar to assigns) so the user doesn't have to repeatedly call `put`